parser.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Linq;
+
+class day4
+{
+    public static string result;
+    //test for global Variable
+    public static void Main()
+    {
+        string url = @"https://www.microsoft.com/en-us/translator/business/languages/
+";
+
+        WebClient client = new WebClient();
+        client.Credentials = CredentialCache.DefaultCredentials;
+        byte[] data = client.DownloadData(url);
+        string html = Encoding.UTF8.GetString(data);
+        //Console.WriteLine(html);
+        //debug
+        List<row> textList = GetContent(html);
+        /*foreach (row line in textList) { 
+            Console.WriteLine(line.NeuralTranslation);
+        }*/
+        //debug
+        using (var writer = new StreamWriter("/Users/admin/Desktop/practice/result.csv"))
+        {
+            foreach (row each in textList)
+            {
+                var line = string.Format("{0},{1},{2},{3},{4},{5}", each.TranslatorLanguage, each.NeuralTranslation, each.Customization, each.Transliteration, each.Dictionary, each.SpeechTranslation);
+                writer.WriteLine(line);
+                writer.Flush();
+            }
+        }
+    }
+
+    public struct row
+    {
+        public string TranslatorLanguage;
+        public bool NeuralTranslation;
+        public bool Customization;
+        public bool Transliteration;
+        public bool Dictionary;
+        public bool SpeechTranslation;
+    }
+    public static List<row> GetContent(string str)
```

Comment:
The method is good and concise. Here is a suggestion that you can have a try:

1. Make the return type of "GetContent" from List<row> to IEnumerable<row>
2. For each iteration of "trCollection", yield return your row instance
3. In main method, use "foreach" to iterate the returned IEnumerable object in foreach

This will get rid of storing all row instances in a collection, so that it will optimize the memory use a little bit. Note that, elements in IEnumerable cannot be modified because they are immutable, and struct instance cannot be modified either (it is value type). If you have any further need of modifying each element with some specific operations, you can still consider return it as a list.